src/js-api/handlers/ls.ts

```diff
@@ -0,0 +1,125 @@
+import { createBaseplateFetch, exitWithError } from "../../cli-utils";
+import {
+  EndpointGetCustomerWebAppsResBody,
+  EndpointGetEnvironmentsResBody,
+  EndpointGetMicrofrontendsResBody,
+  EndpointGetMyCustomerOrgResBody,
+} from "@baseplate-sdk/web-app";
+import { logTable, warning } from "../../cli-logger";
+
+export async function list(args: ListArgs) {
+  const baseplateFetch = createBaseplateFetch(args);
+  const { id: customerOrgId, orgKey } =
+    await baseplateFetch<EndpointGetMyCustomerOrgResBody>(`/api/orgs/me`);
+  const resourceLister = new ResourceLister(orgKey, customerOrgId);
+  await resourceLister.list(args);
+}
+
+export interface ListArgs {
+  baseplateToken: string;
+  resource: "mfe" | "webapp" | "env";
+}
+
+export class ResourceLister {
+  constructor(
+    private readonly orgKey: string,
+    private readonly customerOrgId: string,
+  ) {}
+
+  async list(args: ListArgs) {
+    switch (args.resource) {
+      case "mfe":
+        await this.listMicrofrontends(args);
+        break;
+      case "webapp":
+        await this.listWebapps(args);
+        break;
+      case "env":
+        await this.listEnvironments(args);
+        break;
+      default:
+        exitWithError(
+          `Unknown resource "${args.resource}", ` +
+            `valid resources are: ${warning(" mfe | webapp | env ")}`,
+        );
+    }
+  }
+
+  async listMicrofrontends(args: ListArgs) {
+    const { microfrontends } = await createBaseplateFetch(
+      args,
+    )<EndpointGetMicrofrontendsResBody>(
+      `/api/orgs/${this.customerOrgId}/microfrontends`,
+    );
+    logTable(
+      microfrontends.map((m) => MicrofrontendTableRow.parse(m, this.orgKey)),
+    );
+  }
+
+  async listEnvironments(args: ListArgs) {
+    const { environments } = await createBaseplateFetch(
+      args,
+    )<EndpointGetEnvironmentsResBody>(
+      `/api/orgs/${this.customerOrgId}/environments`,
+    );
+    logTable(environments.map((m) => EnvironmentTableRow.parse(m)));
+  }
+
+  async listWebapps(args: ListArgs) {
+    const { customerWebApps } = await createBaseplateFetch(
```

Comment:
Why recreate baseplateFetch multiple times, in each method?

Reply:
yeah I should fix this, most likely I just copied it sine `createBaseplateFetch` is not typed so it's hard to carry the same instance around but will fix it